Users should be able to define a  `manualScale` instead of being locked into the `mainScreen.scale`

regnerjr opened the pull request:
In order to stay flexible, users of this library "need" to upload two version of there reference images to keep tests passing on various environments (Other Developers, CI, etc)

This proposal would not lock users in to always using the "one true simulator" to run their tests on.

We've recently adopted Snapshot testing at my shop and I don't really want to be the the code review czar ensuring every new PR which adds a snapshot has one at `@2x` and `@3x`. Other cancers are the size of our git repo growing over time to support 2 screenshots for each new test.

>Currently the framework uses [[UIScreen mainScreen] scale] for saving and comparing snapshots of views and layers.
>
>If the view or layer are meant to be displayed at a different scale that doesn't match the main screen's one, the screenshots get resized (minified / magnified) and the comparisons are not correct any more from the point of view of the user, since the images are different from what will be shown to the user.
>
>This PR tries to remove this limitation for those special cases by allowing the test developer to manually override the scale that will be used.

*******************
This work was originally proposed by @djrtl as https://github.com/facebookarchive/ios-snapshot-test-case/pull/166
*******

Thanks for doing the original implementation @djrtl, I hope I did not overstep by re-opening this here now that this lib is under new ownership.

MatteoMecucci-TomTom committed:
Add support for using a scale different from the main screen's one.

BorysZhdanov-TomTom committed:
Fixing manualScale bug
* Manual scale wasn't propagated properly to image capture and reference image selection methods

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/uber/ios-snapshot-test-case?pullRequest=15) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/uber/ios-snapshot-test-case?pullRequest=15) before we can accept your contribution.<br/>**0** out of **3** committers have signed the CLA.<br/><br/>:x: BorysZhdanov-TomTom<br/>:x: regnerjr<br/>:x: MatteoMecucci-TomTom<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/uber/ios-snapshot-test-case?pullRequest=15) it.</sub>

regnerjr committed:
Change type of `manualScale` to `CGFloat`

This better reflects the duality between it an `UIScreen.scale` which
has type `CGFloat`

djrtl commented:
@regnerjr of course no problem, go ahead 👍

alanzeino commented:
I'll take a look at this soon.

alanzeino commented:
Is this resolved here?

https://github.com/facebookarchive/ios-snapshot-test-case/pull/166#issuecomment-334257748

m-ruhl commented:
@alanzeino 
The described problem seams to be fixed.
Unfortunately with the scaling the images have most of the time little differences, and are failing therefore.

regnerjr commented:
Thanks for the support on this Alan. 👍 Really glad you are supporting this project.

I'm thinking perhaps a better approach (instead of having a fixed size window and trying to do all of our snapshotting in there) will be to do snapshots for all devices we support. 
(Current project is iPhone only, only Portrait orientation). 
I've been thinking of adopting the technique of running the snapshot tests in their own, test target and making a build for each device and testing all 4 configs iPhoneX, iPhone 5.5", iPhone 4.7", and iPhone 4".

Glad we've got a fix for that issue sorted out (each just needs it's own build folder) https://github.com/uber/ios-snapshot-test-case/issues/37

Cheers. Keep up the good work.

Weizzz commented:
It seems the capability for using `manualScale` is no longer possible and it's back to using `[[UIScreen mainScreen] scale]`.

alanzeino commented:
> It seems the capability for using `manualScale` is no longer possible and it's back to using `[[UIScreen mainScreen] scale]`.

What did you see when debugging the manualScale?
